### PR TITLE
[cluster-test] Convert effects and experiments to async - migrate to tokio 2.0 (2/n)

### PR DESCRIPTION
**Pull request by andll**
This diff migrates core cluster test abstractions such as effects and experiments to use async API.
This diff removes dependency on thread pool executor from cluster test as all parallel jobs are now run on tokio runtime

### CODE REVIEW

File: testsuite/cluster-test/src/effects/mod.rs [hunk 1]
```diff
@@ -9,50 +9,21 @@ mod reboot;
...
 pub use reboot::Reboot;
 pub use remove_network_effects::RemoveNetworkEffects;
-use slog_scope::info;
 use std::fmt::Display;
 pub use stop_container::StopContainer;
+
 pub trait Action: Display + Send {
-    fn apply(&self) -> failure::Result<()>;
-    fn is_complete(&self) -> bool;
+    fn apply(&self) -> BoxFuture<failure::Result<()>>;
```
Comment (ankushagarwal) [inline testsuite/cluster-test/src/effects/mod.rs]: Why not just `fn apply(&self) -> Future<failure::Result<()>>;`

Comment (andll) [reply testsuite/cluster-test/src/effects/mod.rs]: That won't compile :)

**Pull request closed without merging**